workers/common/src/main/kotlin/common/env/GitConfigGenerator.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.common.env
+
+import com.typesafe.config.Config
+
+import org.eclipse.apoapsis.ortserver.config.ConfigManager
+import org.eclipse.apoapsis.ortserver.model.CredentialsType
+import org.eclipse.apoapsis.ortserver.utils.config.getStringOrNull
+import org.eclipse.apoapsis.ortserver.workers.common.env.definition.EnvironmentServiceDefinition
+
+import org.slf4j.LoggerFactory
+
+/**
+ * A specialized generator class to generate a Git configuration file.
+ */
+class GitConfigGenerator(val configManager: ConfigManager) : EnvironmentConfigGenerator<EnvironmentServiceDefinition> {
+    companion object {
+        private const val GIT_CONFIG_FILE_NAME = ".gitconfig"
+
+        private val logger = LoggerFactory.getLogger(GitConfigGenerator::class.java)
+
+        // Create url.<base>.insteadOf sections in the Git configuration file.
+        // See https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf
+        internal const val GIT_CONFIG_URL_INSTEAD_OF = "gitConfigUrlInsteadOf"
+
+        internal fun parseGitConfigUrlInsteadOf(config: String) =
+            config.split(",")
+                .filter { it.contains("=") }
```

Comment:
Probably a warning should be logged if syntactically incorrect declarations are found. Otherwise, it is hard to diagnose problems with the configuration.